ros_github_scripts/ci_for_pr.py

```diff
@@ -394,11 +394,17 @@ def main():
     if parsed.packages:
         packages_changed = ' '.join(parsed.packages)
         if parsed.only_fixes_test:
-            extra_build_args += f' --packages-up-to {packages_changed}'
-            extra_test_args += f' --packages-select {packages_changed}'
+            extra_build_args = f' --packages-up-to {packages_changed}'
+            extra_test_args = f' --packages-select {packages_changed}'
         else:
-            extra_build_args += f' --packages-above-and-dependencies {packages_changed}'
-            extra_test_args += f' --packages-above {packages_changed}'
+            if parsed.colcon_build_args:
+                extra_build_args += f' {packages_changed}'
+            else:
+                extra_build_args = f' --packages-above-and-dependencies {packages_changed}'
+            if parsed.colcon_test_args:
+                extra_test_args += f' {packages_changed}'
+            else:
+                extra_test_args = f' --packages-above {packages_changed}'
```

Comment:
If the behavior is being changed like this, then we must at least add a note to the help string for `--colcon-build-args`. Like "if specified, _must_ have a package selection argument as the final element". Otherwise, with this behavior, if a user says (for example)

```
--colcon-build-args="--continue-on-error"
```

Then the command will be incorrect because the ` {packages_changed}` on the end will not be attached to an option that takes those arguments.

I think the goal here is to allow the user to choose which package selector they want? Maybe it would be better to add a separate option `--package-selector=[packages-up-to, packages-above-and-dependencies, etc...]` so that there can be no "user error"

Reply:
Ah, right. so user can should be able to set arbitrary arguments with those options, previous behavior makes more sense. i will revert this fix.